### ISSUE TITLE
Train Box and Polygon at once

### ISSUE THREAD

**Issue by dennisans2** [ultralytics/ultralytics#6126]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Write now I've got a dataset with both polygon and bounding box. But it removes one of them when I try to train it. Is there any way to train both of them at once or convert one to another one? If there is, please give me an example code or method. Thanks in advance.

**Comment by jordanesikati** [comment]
@dennisans2  hello, could you provide us with more information regarding the issue?
e.g.: label format: 1  0.471  0.528  0.522  0.432  ...
1 -> Class Index
0.471  0.528  0.522  0.432 -> Bounding Box
Polygon ?

However, if you plan to use the current dataset for instance segmentation, please check the polygon labels.

**Comment by Kayzwer** [comment]
@dennisans2 if you mean training for segmentation (polygon), the head will also provide a bounding box output.

**Comment by glenn-jocher** [comment]
Hello @dennisans2,

In YOLOv8, each object detection architecture has its own output format. For instance, if we talk about training for segmentation tasks which involve polygons, it's important to note that the YOLOv8 architecture will also yield a bounding box output.

Typically, if you have a dataset with both polygon and bounding box annotations, the model does not "remove" one of them. Rather, the standard YOLOv8 network architecture can learn to predict bounding boxes and you should be able to use the polygon annotations for instance segmentation tasks. 

However, the YOLOv8 model cannot directly convert one annotation type to the other. For instance, converting a bounding box to a polygon would involve estimating the shape of an object which goes beyond just object detection and requires instance segmentation capabilities. 

To sum up, while you can't technically "train both at once" in the sense of converting one to the other within the model itself, you can certainly use your dataset with both types of annotations to train a YOLOv8 model for object detection (using the bounding boxes) and instance segmentation (using the polygons). 

I hope this clarifies your question. Please feel free to reach out if you have more queries or need further details. 

Best, 
Glenn

**Comment by dennisans2** [comment]
Great. I'll do that

**Comment by glenn-jocher** [comment]
@jordanesikati hello,

Thank you for acknowledging the given response. Great to hear that you'll proceed with the recommended strategy. If you encounter any issues during the process or have more queries about YOLOv8, feel free to continue this thread or open a new one.

Keep in mind that utilizing your dataset with both bounding box and polygon annotations to train a YOLOv8 model should cater to your requirements distinctly addressing both object detection and instance segmentation tasks. 

I appreciate your patience and persistence in resolving your query. Looking forward to hearing about your progress!

Best,
Glenn Jocher
